README.md

```diff
@@ -1,15 +1,192 @@
+
+- **Terraform AWS Provider Configuration**: Configure the AWS provider in your Terraform project. You can add your AWS access and secret keys as environment variables or use other methods for provider configuration.
+
+- **Terraform Configuration File**: Create a new Terraform configuration file in your project, and use this module by specifying its source.
+
+—
+
+## Usage
+
+To use this module, follow these steps:
+
+1. Clone the repository to your local machine:
+
+   ```shell
+   git clone https://github.com/Ontotext-AD/terraform-aws-graphdb.git
+   ```
+
+2. Navigate to the module directory:
+
+   ```shell
+   cd terraform-aws-graphdb/modules/aws-graphdb-cluster
+   ```
+
+3. Create a new Terraform configuration in your project, and use this module by specifying its source:
+
+   ```hcl
+   module "graphdb_cluster" {
+     source = "../../modules/aws-graphdb-cluster"
+
+     # Define your configuration variables here, such as cluster size, VPC settings, and other parameters.
+
+
+   }
+
+   # Configure the required variables for the module.
+   ```
+
+4. Initialize your Terraform workspace and apply the changes:
+
+  ```shell
+    terraform init
+    terraform plan
+  ```
+
+5. Review and confirm the changes when prompted and apply if everything is okay.
+
+  ```shell
+    terraform apply
+  ```
+
+
+
+
+
+## Configuration
+
+This README provides more detailed information about the important variables and how to configure them for the Terraform AWS GraphDB module. Please ensure that you adapt the configuration to match your specific use case and requirements.
+
+### Important Variables (Inputs)
+
+The following are the important variables you should configure when using this module:
+
+- `vpc_id`: The ID of the Virtual Private Cloud (VPC) where you want to create the AWS GraphDB cluster. This is a required variable.
+
+- `instance_type`: The instance type for the GDB cluster nodes. This should match your performance and cost requirements.
+
+- `cluster_size`: The number of instances in the cluster. Recommended is 3, 5 or 7 in order to have consensus according to the [RAFT algorithm](https://raft.github.io/). 
+
+- `subnet_ids`: A list of subnet IDs in your VPC to place the Neptune instances. Ensure that these subnets have proper route tables and security groups configured.
+
+- `security_group_ids`: A list of security group IDs to associate with the instances for network security.
 
-TBD
 
-## License
 
-This code is released under the Apache 2.0 License. See [LICENSE](LICENSE) for more details.
 
-## Contributing
 
+
+## Examples
+
+This repository includes usage examples in the `examples/` directory, such as:
+
+- [VPC with Multiple Availability Zones](./examples/vpc-with-multiple-az): Demonstrates creating a VPC with multiple Availability Zones and provisioning an AWS GraphDB cluster.
+
+Here's an example of how to configure the module in your Terraform project:
+
+#### Example Configuration
```

Comment:
We should be telling people to create `terraform.tfvars` file instead of directly modifying `variables.tf`.